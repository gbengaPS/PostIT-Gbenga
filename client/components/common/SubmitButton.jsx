import React from 'react';
import PropTypes from 'prop-types';

const SubmitButton = props => (
  <div className="row">
    <input type="submit"
    value={props.submitValue}
    className="btn light-blue darken-4 col m4 offset-m4"
    disabled={props.disabled}
    />
  </div>
);
SubmitButton.propTypes = {
  submitValue: PropTypes.string.isRequired,
  disabled: PropTypes.string,
};
export default SubmitButton;
