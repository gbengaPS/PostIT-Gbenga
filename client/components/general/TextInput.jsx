import React from 'react';
import Proptypes from 'prop-types';

const TextInput = props => (
  <div>
    <div className="input-field">
      <input
      type="text" name={props.name}
      id={props.name} onChange={props.handleChange}
      required className={props.classnames} />
      <label htmlFor={props.name}> {props.description} </label>
    </div>
  </div>
);
TextInput.propTypes = {
  name: Proptypes.string.isRequired,
  handleChange: Proptypes.func.isRequired,
  classnames: Proptypes.string,
  description: Proptypes.string.isRequired,
};
export default TextInput;
