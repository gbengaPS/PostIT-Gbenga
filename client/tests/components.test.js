import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TextInput from '../components/common/TextInput';
import SubmitButton from '../components/common/SubmitButton';

configure({ adapter: new Adapter() });

describe('TextInput component', () => {
  const props = {
    type: 'text',
    name: 'test',
    handleChange: () => {},
    description: 'description',
    className: 'whatever'
  };
  const TextInputWrapper = shallow(<TextInput {...props} />);
  it('should render one div', () => {
    expect(TextInputWrapper.find('div').length).toBe(1);
  });
  it('should render correctly', () => {
    expect(TextInputWrapper).toMatchSnapshot();
  });
});
describe('Submit button', () => {
  const props = {
    submitValue: 'create group'
  };
  const wrapper = shallow(<SubmitButton {...props} />);
  it('should render without exploding', () => {
    expect(wrapper.length).toBe(1);
  });
});
