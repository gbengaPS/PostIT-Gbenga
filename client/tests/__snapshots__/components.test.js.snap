// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    className="whatever"
    description="description"
    handleChange={[Function]}
    name="test"
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className={undefined}
          id="test"
          name="test"
          onChange={[Function]}
          required={true}
          type="text"
        />,
        <label
          htmlFor="test"
        >
           
          description
           
        </label>,
      ],
      "className": "input-field",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "id": "test",
          "name": "test",
          "onChange": [Function],
          "required": true,
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "description",
            " ",
          ],
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "description",
          " ",
        ],
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className={undefined}
            id="test"
            name="test"
            onChange={[Function]}
            required={true}
            type="text"
          />,
          <label
            htmlFor="test"
          >
             
            description
             
          </label>,
        ],
        "className": "input-field",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "test",
            "name": "test",
            "onChange": [Function],
            "required": true,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "description",
              " ",
            ],
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "description",
            " ",
          ],
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
